lab_6_pipeline/pipeline.py

```diff
@@ -18,21 +31,51 @@ def __init__(self, path_to_raw_txt_data: Path):
         """
         Initializes CorpusManager
         """
+        self.path_to_raw_txt_data = path_to_raw_txt_data
+        self.json_files = list(self.path_to_raw_txt_data.glob('*_meta.json'))
+        self.txt_files = list(self.path_to_raw_txt_data.glob('*_raw.txt'))
+        self._validate_dataset()
+        self._storage = {}
+        self._scan_dataset()
 
     def _validate_dataset(self) -> None:
         """
         Validates folder with assets
         """
+        if not self.path_to_raw_txt_data.exists():
+            raise FileNotFoundError('file does not exist')
+        if not self.path_to_raw_txt_data.is_dir():
+            raise NotADirectoryError('path does not lead to directory')
+        if not self.json_files or not self.txt_files:
+            raise EmptyDirectoryError('directory is empty')
+        len_json, len_txt = len(self.json_files), len(self.txt_files)
+        if len_json != len_txt:
+            raise InconsistentDatasetError('number of meta and raw files is not equal')
+        for json, txt in zip(self.json_files, self.txt_files):
+            if not json.stat().st_size or not txt.stat().st_size:
+                raise InconsistentDatasetError("files are empty")
+        self.json_files = list(map(lambda link: link.name, self.json_files))
```

Comment:
it is antipattern: first you define self.json_files as list of Paths then - as list of strings.

Reply:
Done